chromium_src/chrome/installer/setup/install.cc

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/installer/setup/install.h"
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/time/time.h"
+#include "chrome/installer/util/shell_util.h"
+
+/*
+ * Work around crbug.com/331836635 "Double prompt to pin to taskbar on Windows".
+ *
+ * Upstream pins the browser to the taskbar both in mini_installer and during
+ * first run. Our online installer runs mini_installer and then immediately
+ * launches the browser. This makes two "Do you want to pin Brave to your
+ * taskbar?" notifications appear.
+ *
+ * The solution implemented here only pins on first run if the browser
+ * installation is more than 3 minutes old. This avoids the second notification
+ * in the scenario above, while still giving non-admin users in system-wide
+ * installations a chance to pin.
+ */
+
+namespace {
+constexpr int kPinAfterMins = 3;
+}
+
+#define set_pin_to_taskbar(...)                                         \
+  set_pin_to_taskbar(ShouldPinToTaskbar(do_not_create_taskbar_shortcut, \
+                                        shortcut_operation, target))
+
+namespace installer {
+
+bool ShouldPinToTaskbar(bool do_not_create_taskbar_shortcut,
```

Comment:
Could this function be in the anonymous namespace above or is this necessary for some reason?

Reply:
Good point. Resolved in 859b7153046a6fb0fdfcb9eca2c6c879c8a78a78.